src/components/oscal-utils/OSCALRestUtils.js

```diff
@@ -46,6 +46,31 @@ export const oscalObjectTypes = {
   },
 };
 
+/**
+ * Populates a map of oscal object types to oscal objects retrieved from a get request.
+ *
+ * @param {string} backendUrl base url of server to fetch from
+ * @param {Object} oscalObjectType Object that contains information on an oscalObject
+ * @param {function} setOscalObjects Function to map a fetched result to the json root name of an oscal object
+ */
+export function populateOscalObjects(
+  backendUrl,
+  oscalObjectType,
+  setOscalObjects
+) {
+  fetch(`${backendUrl}/${oscalObjectType.restPath}`)
+    .then((response) => {
+      if (!response.ok) throw new Error(response.status);
+      else return response.json();
+    })
+    .then((result) => {
+      setOscalObjects((current) => ({
+        ...current,
+        [oscalObjectType.jsonRootName]: result,
+      }));
+    });
+}
```

Comment:
Hmm this feels a little too specific to one use case to be a shared REST util. It depends on knowing a specific structure for the underling data type under `setOscalObjects` and what it looks like (and we don't document that expectation).

I think I'd rather see an interface like the following (I've written it in TypeScript to make the intent a little more clear). Though I've removed error handling in this example to make it a bit clearer.

```ts
interface OscalObjectType {
  restPath: string;
  jsonRootName: string;
}

export function fetchAllResourcesOfType(
  backendUrl: string,
  oscalObjectType: OscalObjectType,
  handleResult: (any) => void,
): void {
  fetch(`${backendUrl}/${oscalObjectType.restPath}`)
    .then((response) => response.json())
    .then((result) => handleResult(result));
}

oscalObjectTypes.forEach((oscalObjectType) => {
  fetchAllResourcesOfType(backEndUrl, oscalObjectType, (result) => {
    setOscalObjects((current) => ({
      ...current,
      [oscalObjectType.jsonRootName]: result,
    });
  });
}
```

Reply:
This also feels a bit more testable eventually too (because you separate the responsibility for fetching the data from the responsibility for editing state). One could imagine a `handleResult` that supports testing instead.